### PR TITLE
[Quality] Adds checkstyle task to Android project

### PR DESCRIPTION
**Pull request by sleroux**
Copies over the checkstyle rules and task we use for other Shopify Android projects:

https://github.com/Shopify/mobile-shopify/wiki/Checkstyle

It's also worth adding this to Android Studio directly so you can run the rules as you develop:

https://github.com/Shopify/mobile-shopify/wiki/Checkstyle-in-Android-Studio

### CODE REVIEW

File: unity_buy_android/checkstyle/config.xml [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+                <property name="tokens" value="BAND, BOR, BSR, BXOR, DIV, EQUAL, GE, GT, LAND, LE, LITERAL_INSTANCEOF, LOR, LT, MINUS, MOD, NOT_EQUAL, PLUS, QUESTION, SL, SR, STAR "/>
+            </module>
+                    <module name="EmptyCatchBlock">
+                <property name="exceptionVariableName" value="expected"/>
+            </module>
+
+                <module name="LocalFinalVariableName"/>
+        <module name="StaticVariableName"/>
+        <module name="ConstantName"/>
+        -->
```
Comment (dengjeffrey) [inline unity_buy_android/checkstyle/config.xml]: There's a lot of commented out lines here, will we need these lines? If not should they be removed?

File: scripts/test_android_unit.sh [hunk 2]
```diff
@@ -2,5 +2,8 @@
 
 ANDROID_PLUGIN_ROOT="$PROJECT_ROOT/unity_buy_android"
 
+# Run checkstyle before running tests
+"$ANDROID_PLUGIN_ROOT"/gradlew unity_buy_plugin:checkstyle -p "$ANDROID_PLUGIN_ROOT" || die "Failed to run checkstyle"
+
```
Comment (sav007) [inline scripts/test_android_unit.sh]: You can do it inside of gradle, so that you don't need this line.

```
afterEvaluate {
      if (project.tasks.findByName('check')) {
        check.dependsOn('checkstyle')
      }
    }
```

Comment (sleroux) [inline scripts/test_android_unit.sh]: Should I also be running the `check` task instead of `testDebug` for running tests? Will that run unit and instrument tests?

File: unity_buy_android/checkstyle/config.xml (hunk 1 above)
Comment (sleroux) [inline unity_buy_android/checkstyle/config.xml]: I'm just going to leave it as is since this is just a copy of the Shopify Android is using.

File: unity_buy_android/unity_buy_plugin/src/main/java/com/shopify/unity/buy/UnityBuyPlugin.java [hunk 3]
```diff
@@ -1,6 +1,8 @@
 package com.shopify.unity.buy;
 
 public class UnityBuyPlugin {
+    private UnityBuyPlugin() { }
```
Comment (dengjeffrey) [inline unity_buy_android/unity_buy_plugin/src/main/java/com/shopify/unity/buy/UnityBuyPlugin.java]: Quick question, why is this needed?

Comment (sleroux) [inline unity_buy_android/unity_buy_plugin/src/main/java/com/shopify/unity/buy/UnityBuyPlugin.java]: Checkstyle complains that utility classes shouldn't have a default or public constructors. Makes sense considering a class that only has static methods should never be instantiated. 

This class it just a dummy but figured I'd fix it so the test script doesn't fail.

Comment (sav007) [inline unity_buy_android/unity_buy_plugin/src/main/java/com/shopify/unity/buy/UnityBuyPlugin.java]: Even more it must be final. Any Util class must be final, with private constructor.

File: scripts/test_android.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+. $(dirname $0)/common.sh
+
+ANDROID_PLUGIN_ROOT="$PROJECT_ROOT/unity_buy_android"
+
+"$ANDROID_PLUGIN_ROOT"/gradlew unity_buy_plugin:check -p "$ANDROID_PLUGIN_ROOT" || die "Failed to check Android plugin"
```
Comment (sav007) [inline scripts/test_android.sh]: If I'm not wrong but check do testing as well

Comment (sleroux) [inline scripts/test_android.sh]: Ah yup looks like it does! I'll remove the redundancy

**Comment by sleroux** [comment]
@sav007 Removed the redundant gradle call. 👍?
